src/app/shared/components/course-form/course-form.component.ts

```diff
@@ -49,26 +49,25 @@ export class CourseFormComponent implements OnInit, OnDestroy {
 
   createInit() {
     this.buildForm();
-    this.subscription = this.courseStore.authors$.subscribe(authors$ => {
-      this.allAuthors = authors$;
-      this.nonCourseAuthors = this.allAuthors.filter(
-        authorAllElement =>
-          !this.authors.value.some((authorElement: Author) => authorElement.id === authorAllElement.id)
-      );
-    });
+    this.subscribeToAllAuthors();
```

Comment:
Okay, this is much better. However there is still place to make the code shorter.
In OnInit, based on the isUpdate prop you are triggering some functions. Both updateInint and createInit calls the subscribe and buildForm functions. They do look alike - however the order of operations are different.

Reply:
Right! no need to separate inits.